include/openmc/hdf5_interface.h

```diff
@@ -333,7 +334,9 @@ write_attribute(hid_t obj_id, const char* name, const std::vector<T>& buffer)
 // Templates/overloads for write_dataset
 //==============================================================================
 
-template<typename T> inline void
+// Template for scalars (ensured by SFINAE)
+template<typename T> inline
+std::enable_if_t<std::is_scalar<std::decay_t<T>>::value>
 write_dataset(hid_t obj_id, const char* name, T buffer)
```

Comment:
I find it confusing that the return type isn't explicit with this function.  Can you easily switch the `enable_if_t` to operate through a template parameter rather than the return type?

Reply:
Yeah, there's a hidden default void in `std::enable_if_t`. I actually tried to do `enable_if_t` as template parameter, but for some reason that didn't work. This is my first time using this technique, so it's possible I did something wrong. If you know of a way to get it to work via a template parameter, I'm all ears.

Reply:
Hmm, I couldn't figure out the template parameter way either.  Oh well, it's only a small issue.